### PR TITLE
issue/viewpager_indicator

### PR DESCRIPTION
**Pull request by nbradbury**
This PR moves the view pager indicator out of the login prologue and into its own re-usable widget. The reason for this is the [payment tutorial](https://github.com/woocommerce/woocommerce-android/issues/4341) requires a similar indicator and it made sense to make the existing widget re-usable.

To test, simply step through the login prologue and verify the swipeable view pager works as expected. I recommend testing on a real device as view pagers can be flaky on emulators.

Screenshot_20210706_095424

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt [hunk 1]
```diff
@@ -52,11 +52,12 @@ class LoginPrologueFragment : androidx.fragment.app.Fragment(R.layout.fragment_l
...
         })
 
-        binding.viewPagerIndicator.setListener(object : OnIndicatorClickedListener {
+        val listener = object : OnIndicatorClickedListener {
             override fun onIndicatorClicked(index: Int) {
                 binding.viewPager.currentItem = index
             }
-        })
+        }
+        binding.viewPagerIndicator.setupFromViewPager(binding.viewPager, listener)
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt]: What do you think if we remove `listener` from `setupFromViewPager` signature, and internally, inside `WCViewPageIndicator`. Also, we could actually move the following code:

```
        binding.viewPager.addOnPageChangeListener(object : ViewPager.SimpleOnPageChangeListener() {
            override fun onPageSelected(position: Int) {
                binding.viewPagerIndicator.setSelectedIndicator(position)
            }
        })
```

Inside `WCViewPagerIndicator` like that:

```
class WCViewPagerIndicator @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0,
    defStyleRes: Int = 0
) : LinearLayout(context, attrs, defStyleAttr, defStyleRes) {
    private val indicators = ArrayList<ImageView>()
    private var pageCount: Int = 0

    fun setupFromViewPager(viewPager: ViewPager) {
        this.pageCount = viewPager.adapter?.count ?: 0

        orientation = HORIZONTAL
        val padding = context.resources.getDimensionPixelSize(R.dimen.margin_small)

        for (page in 0 until pageCount) {
            ImageView(context).also { imageView ->
                imageView.setImageResource(R.drawable.ic_tab_indicator)
                imageView.isSelected = page == 0
                imageView.setPadding(padding, 0, padding, 0)
                this.addView(imageView)
                indicators.add(imageView)

                imageView.setOnClickListener {
                    viewPager.currentItem = page
                }
            }
        }

        viewPager.addOnPageChangeListener(object : ViewPager.SimpleOnPageChangeListener() {
            override fun onPageSelected(position: Int) {
                setSelectedIndicator(position)
            }
        })
    }

    private fun setSelectedIndicator(index: Int) {
        for (i in 0 until pageCount) {
            indicators[i].isSelected = i == index
        }
    }
}
```

I think it's quite reasonable to expect that `PageIndicator` that works with `ViewPager`, will select indicators and move pages out of the box without need of seting it up everytime.

This way we, also increase code reusability when we will use the same page indicator for the case of payment tutorial

**Review by kidinov — CHANGES_REQUESTED** [review]
Works and looks good! I would just consider to move some logic inside of the PagerIndicator so it can be more like a "plug-and-play" component

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt (hunk 1 above)
Comment (nbradbury) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt]: @kidinov That is a **much** cleaner approach, thanks for pointing it out. I made that change in a3a3209.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/widgets/WCViewPagerIndicator.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        viewPager.addOnPageChangeListener(object : ViewPager.SimpleOnPageChangeListener() {
+            override fun onPageSelected(position: Int) {
+                setSelectedIndicator(position)
+            }
+        })
+    }
+
+    fun setSelectedIndicator(index: Int) {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/widgets/WCViewPagerIndicator.kt]: I think we could make it `private`, wdyt?

**Review by kidinov — APPROVED** [review]
Looks good now! There is a small comment left. Please go ahead with merging if you think that it makes sense to make it private or not

File: WooCommerce/src/main/kotlin/com/woocommerce/android/widgets/WCViewPagerIndicator.kt (hunk 2 above)
Comment (nbradbury) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/widgets/WCViewPagerIndicator.kt]: Done in dc9ecf3.

**Pull request merged**